app/components/search/CategoriesRefinementList.jsx

```diff
@@ -13,68 +14,61 @@ class CategoriesRefinementList extends Component {
     super(props);
     this.changeRefinement = this.changeRefinement.bind(this);
     this.setChecks = this.setChecks.bind(this);
-    this.categoriesMapping = {
-      'Basic Needs & Shelter': ['Basic Needs & Shelter'],
-      'Eviction Prevention': ['Eviction Prevention'],
-      'Health & Medical': ['Health & Medical'],
-      Housing: ['Housing'],
-      Legal: ['Legal'],
-      Employment: ['Employment'],
-    };
     const checks = this.setChecks();
     this.state = {
       isChecked: checks,
     };
   }
 
   componentDidUpdate(prevProps) {
-    // Typical usage (don't forget to compare props):
     const { currentRefinement } = this.props;
-    if (currentRefinement !== prevProps.currentRefinement) {
+    if (currentRefinement.sort().join(',') !== prevProps.currentRefinement.sort().join(',')) {
       const checks = this.setChecks();
       // setState is done in a condition so it won't create loop
       this.setState({ isChecked: checks }); // eslint-disable-line react/no-did-update-set-state
     }
   }
 
   setChecks() {
-    const { currentRefinement } = this.props;
-    const mapKeys = Object.keys(this.categoriesMapping);
+    const mapKeys = Object.keys(categoriesMapping);
     const checks = [];
-    for (let i = 0; i < mapKeys.length; i++) {
-      const key = mapKeys[i];
-      let atLeastOneRefined = false;
-      for (let i_1 = 0; i_1 < this.categoriesMapping[key].length; i_1++) {
-        const val = this.categoriesMapping[key][i_1];
-        if (currentRefinement.includes(val)) {
-          atLeastOneRefined = true;
-          break;
-        }
-      }
-      checks[key] = atLeastOneRefined;
-    }
+    mapKeys.forEach(key => {
+      checks[key] = this.keyHasAtLeastOneRefined(key);
```

Comment:
I see that this was already like this before, but `checks` is declared as an array, but for some reason we are treating it as a key-value mapping. It's probably clearer to declare it as `checks = {}`.